In [50]:
import pandas as pd
import os
import numpy as np

In [11]:
col_names = ["Location", "Age Group", "TimeFrame", "DataFormat", "Data", "Fips"]
ervisits = pd.read_csv('ervisits.csv', names=col_names, skiprows=9, )
ervisits


Location      Age Group  TimeFrame DataFormat  \
0                  Bayside - Little Neck  5 to 17 Years       2005       Rate   
1                  Bayside - Little Neck  0 to 17 Years       2005     Number   
2     Bedford Stuyvesant - Crown Heights   0 to 4 years       2005       Rate   
3     Bedford Stuyvesant - Crown Heights   0 to 4 years       2005     Number   
4     Bedford Stuyvesant - Crown Heights  5 to 17 Years       2005     Number   
...                                  ...            ...        ...        ...   
2872                         Willowbrook   0 to 4 years       2016     Number   
2873                         Willowbrook   0 to 4 years       2016       Rate   
2874                         Willowbrook  5 to 17 Years       2016     Number   
2875                         Willowbrook  5 to 17 Years       2016       Rate   
2876                         Willowbrook  0 to 17 Years       2016     Number   

        Data    Fips  
0       31.1  uhf404  
1       81.0  uhf404  
2      686.7  uhf203  
3     1520.0  uhf203  
4     1785.0  uhf203  
...      ...     ...  
2872    35.0  uhf503  
2873    75.0  uhf503  
2874    58.0  uhf503  
2875    44.5  uhf503  
2876    93.0  uhf503  

[2877 rows x 6 columns]

In [12]:
ervisits['Fips'] = ervisits['Fips'].str[3:]
ervisits

Location      Age Group  TimeFrame DataFormat  \
0                  Bayside - Little Neck  5 to 17 Years       2005       Rate   
1                  Bayside - Little Neck  0 to 17 Years       2005     Number   
2     Bedford Stuyvesant - Crown Heights   0 to 4 years       2005       Rate   
3     Bedford Stuyvesant - Crown Heights   0 to 4 years       2005     Number   
4     Bedford Stuyvesant - Crown Heights  5 to 17 Years       2005     Number   
...                                  ...            ...        ...        ...   
2872                         Willowbrook   0 to 4 years       2016     Number   
2873                         Willowbrook   0 to 4 years       2016       Rate   
2874                         Willowbrook  5 to 17 Years       2016     Number   
2875                         Willowbrook  5 to 17 Years       2016       Rate   
2876                         Willowbrook  0 to 17 Years       2016     Number   

        Data Fips  
0       31.1  404  
1       81.0  404  
2      686.7  203  
3     1520.0  203  
4     1785.0  203  
...      ...  ...  
2872    35.0  503  
2873    75.0  503  
2874    58.0  503  
2875    44.5  503  
2876    93.0  503  

[2877 rows x 6 columns]

In [32]:
#Filtering To include only data from 2015 and 2016
ervisits2015_2016 = ervisits[(ervisits['TimeFrame'] == 2015) 
                             | (ervisits['TimeFrame'] == 2016)]
ervisits2015_2016

Location      Age Group  TimeFrame DataFormat  Data Fips
2397  Bayside - Little Neck   0 to 4 years       2015       Rate  59.9  404
2398  Bayside - Little Neck   0 to 4 years       2015     Number  28.0  404
2399  Bayside - Little Neck  5 to 17 Years       2015     Number  50.0  404
2400  Bayside - Little Neck  5 to 17 Years       2015       Rate  39.0  404
2401  Bayside - Little Neck  0 to 17 Years       2015     Number  78.0  404
...                     ...            ...        ...        ...   ...  ...
2872            Willowbrook   0 to 4 years       2016     Number  35.0  503
2873            Willowbrook   0 to 4 years       2016       Rate  75.0  503
2874            Willowbrook  5 to 17 Years       2016     Number  58.0  503
2875            Willowbrook  5 to 17 Years       2016       Rate  44.5  503
2876            Willowbrook  0 to 17 Years       2016     Number  93.0  503

[480 rows x 6 columns]

In [40]:
## Filtering Data to include all children from 0 to 17 years old
ervisits2015_2016_0to17 = ervisits2015_2016[(ervisits2015_2016['Age Group'] == '0 to 17 Years') 
                             & (ervisits2015_2016['DataFormat'] == 'Number')]
ervisits2015_2016_0to17.dtypes
ervisits2015_2016_0to17['Fips'] = ervisits2015_2016_0to17['Fips'].astype(int)
ervisits2015_2016_0to17.dtypes

/Users/tajahneiwilliams/opt/anaconda3/lib/python3.7/site-packages/ipykernel_launcher.py:5: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  """


Location       object
Age Group      object
TimeFrame       int64
DataFormat     object
Data          float64
Fips            int64
dtype: object

In [42]:
fipscombined_df = ervisits2015_2016_0to17.merge(uhf_zipcode_crosswalk_df, how='left', on='Fips')
fipscombined_df

Location      Age Group  TimeFrame DataFormat  \
0                Bayside - Little Neck  0 to 17 Years       2015     Number   
1   Bedford Stuyvesant - Crown Heights  0 to 17 Years       2015     Number   
2              Bensonhurst - Bay Ridge  0 to 17 Years       2015     Number   
3                         Borough Park  0 to 17 Years       2015     Number   
4                                Bronx  0 to 17 Years       2015     Number   
..                                 ...            ...        ...        ...   
91                     Upper West Side  0 to 17 Years       2016     Number   
92         Washington Heights - Inwood  0 to 17 Years       2016     Number   
93                         West Queens  0 to 17 Years       2016     Number   
94             Williamsburg - Bushwick  0 to 17 Years       2016     Number   
95                         Willowbrook  0 to 17 Years       2016     Number   

       Data  Fips        Borough                            NTA_Name  \
0      78.0   404         Queens               Bayside - Little Neck   
1    3040.0   203       Brooklyn  Bedford Stuyvesant - Crown Heights   
2     193.0   209       Brooklyn             Bensonhurst - Bay Ridge   
3     485.0   206       Brooklyn                        Borough Park   
4   17535.0     5            NaN                                 NaN   
..      ...   ...            ...                                 ...   
91    458.0   304      Manhattan                     Upper West Side   
92   1213.0   301      Manhattan         Washington Heights - Inwood   
93   1771.0   402         Queens                         West Queens   
94   1899.0   211       Brooklyn             Williamsburg - Bushwick   
95     93.0   503  Staten Island                         Willowbrook   

                                            Zipcode  
0                        11361, 11362, 11363, 11364  
1                 11213, 11212, 11216, 11233, 11238  
2                               11209, 11214, 11228  
3                        11204, 11218, 11219, 11230  
4                                               NaN  
..                                              ...  
91                              10023, 10024, 10025  
92                10031, 10032, 10033, 10034, 10040  
93  11368, 11369, 11370, 11372, 11373, 11377, 11378  
94                              11206, 11221, 11237  
95                                            10314  

[96 rows x 9 columns]

def set_zipcode(row):
    if row["Fips"] == "uhf101":
        return "11361, 11362, 11363, 11364, 11365, 11366, 11367"
    elif row["Fips"] == "uhf102":
        return "10466, 10469, 10470, 10475"
    elif row["Fips"] == "uhf103":
        return "10458, 10467, 10468"
    elif row["Fips"] == "uhf104":
        return "10461, 10462, 10464, 10465, 10472, 10473"
    elif row["Fips"] == "uhf105":
        return "10451, 10452, 10453, 10454, 10455, 10456, 10457, 10459, 10460, 10461"
    elif row["Fips"] == "uhf106":
        return "10451, 10452, 10453, 10454, 10455, 10456, 10457, 10459, 10460, 10461"
    elif row["Fips"] == "uhf107":
        return "10451, 10452, 10453, 10454, 10455, 10456, 10457, 10459, 10460, 10461"
    elif row["Fips"] == "uhf201":
        return "11211, 11222"
    elif row["Fips"] == "uhf202":
        return "11201, 11205, 11215, 11217, 11231"
    elif row["Fips"] == "uhf203":
        return "11212, 11213, 11216, 11233, 11238"
    elif row["Fips"] == "uhf204":
        return "11207, 11208"
    elif row["Fips"] == "uhf205":
        return "11220, 11232"
    elif row["Fips"] == "uhf206":
        return "11204, 11218, 11219, 11230"
    elif row["Fips"] == "uhf207":
        return "11203, 11210, 11225, 11226"
    elif row["Fips"] == "uhf208":
        return "11234, 11236, 11239"
    elif row["Fips"] == "uhf209":
        return "11209, 11214, 11228"
    elif row["Fips"] == "uhf210":
        return "11223, 11224, 11229, 11235"
    elif row["Fips"] == "uhf211":
        return "11206, 11221, 11237"
    elif row["Fips"] == "uhf301":
        return "10031, 10032, 10033, 10034, 10040"
    elif row["Fips"] == "uhf302":
        return "10026, 10027, 10030, 10037, 10039"
    elif row["Fips"] == "uhf303":
        return "10029, 10035"
    elif row["Fips"] == "uhf304":
        return "10023, 10024, 10025, 10069"
    elif row["Fips"] == "uhf305":
        return "10010, 10016, 10017, 10021, 10022, 10028, 10044, 10065, 10075, 10128, 10162, 10165, 10170, 10171"
    elif row["Fips"] == "uhf306":
        return "10001, 10011, 10012, 10013, 10014, 10018, 10019, 10020, 10036"
    elif row["Fips"] == "uhf307":
        return "10010, 10016, 10017, 10021, 10022, 10028, 10044, 10065, 10075, 10128, 10162, 10165, 10170, 10171"
    elif row["Fips"] == "uhf308":
        return "10001, 10011, 10012, 10013, 10014, 10018, 10019, 10020, 10036"
    elif row["Fips"] == "uhf309":
        return "10002, 10003, 10004, 10005, 10006, 10007, 10009, 10038, 10048, 10280, 10282"
    elif row["Fips"] == "uhf310":
        return "10002, 10003, 10004, 10005, 10006, 10007, 10009, 10038, 10048, 10280, 10282"
    elif row["Fips"] == "uhf401":
        return "11101, 11102, 11103, 11104, 11105, 11106, 11109"
    elif row["Fips"] == "uhf402":
        return "11368, 11369, 11370, 11372, 11373, 11377, 11378"
    elif row["Fips"] == "uhf403":
        return "11354, 11355, 11356, 11357, 11358, 11359, 11360"
    elif row["Fips"] == "uhf404":
        return "11361, 11362, 11363, 11364, 11365, 11366, 11367"
    elif row["Fips"] == "uhf405":
        return "11374, 11375, 11379, 11385"
    elif row["Fips"] == "uhf406":
        return "11361, 11362, 11363, 11364, 11365, 11366, 11367"
    elif row["Fips"] == "uhf407":
        return "11414, 11415, 11416, 11417, 11418, 11419, 11420, 11421"
    elif row["Fips"] == "uhf408":
        return "11412, 11423, 11430, 11432, 11433, 11434, 11435, 11436"
    elif row["Fips"] == "uhf409":
        return "11001, 11004, 11005, 11040, 11411, 11413, 11422, 11426, 11427, 11428, 11429"
    elif row["Fips"] == "uhf410":
        return "11691, 11692, 11693, 11694, 11697"
    elif row["Fips"] == "uhf501":
        return "10301, 10302, 10303, 10304, 10305, 10310"
    elif row["Fips"] == "uhf502":
        return "10301, 10302, 10303, 10304, 10305, 10310"
    elif row["Fips"] == "uhf503":
        return "10306, 10307, 10308, 10309, 10312, 10314"
    elif row["Fips"] == "uhf504":
        return "10306, 10307, 10308, 10309, 10312, 10314"
    else:
        return "N/A"

ervisits = ervisits.assign(zipcode=ervisits.apply(set_zipcode, axis=1))

ervisits

In [6]:
ervisits.to_csv(r'/Users/tajahneiwilliams/Documents/Project2/static/data/ervisits_zipcodes.csv', index=False)

In [55]:
zipcodes_df = pd.read_csv('nycZipCodes.csv')


In [53]:
zipcodes_df.dropna(axis=1, how="any", inplace=False)

geometry                                        coordinates  \
0    MultiPolygon  -73.840759,40.625364,-73.843062,40.627933,-73....   
1    MultiPolygon  -73.915441,40.875591,-73.915087,40.876496,-73....   
2    MultiPolygon  -73.932131,40.869451,-73.931277,40.868844,-73....   
3    MultiPolygon  -73.837608,40.743238,-73.835162,40.743338,-73....   
4    MultiPolygon  -73.880111,40.837234,-73.877472,40.836564,-73....   
..            ...                                                ...   
257  MultiPolygon  -73.974075,40.759256,-73.973895,40.759507,-73....   
258  MultiPolygon  -73.979243,40.762827,-73.97953,40.762948,-73.9...   
259  MultiPolygon  -73.986732,40.748571,-73.986554,40.74881,-73.9...   
260  MultiPolygon  -73.978695,40.755496,-73.978479,40.755407,-73....   
261  MultiPolygon  -73.951328,40.76931,-73.951652,40.769446,-73.9...   

     postalcode       po_name state    borough  st_fips  cty_fips  \
0         11693  Far Rockaway    NY   Brooklyn       36        47   
1         10463         Bronx    NY  Manhattan       36        61   
2         10033      New York    NY  Manhattan       36        61   
3         11367      Flushing    NY     Queens       36        81   
4         10472         Bronx    NY      Bronx       36         5   
..          ...           ...   ...        ...      ...       ...   
257       10055      New York    NY  Manhattan       36        61   
258       10105      New York    NY  Manhattan       36        61   
259       10118      New York    NY  Manhattan       36        61   
260       10176      New York    NY  Manhattan       36        61   
261       10162      New York    NY  Manhattan       36        61   

       shape_leng    shape_area  \
0     9479.917277  3.497516e+06   
1     7791.517127  3.119702e+06   
2    29415.607123  1.615605e+07   
3    33506.126605  7.281015e+07   
4    27006.042411  3.096325e+07   
..            ...           ...   
257    842.265584  3.722354e+04   
258    769.015786  3.542216e+04   
259   1392.546380  9.173372e+04   
260    590.377333  1.828909e+04   
261    624.792615  2.103489e+04   

                                                    id  
0    http://nyc.pediacities.com/Resource/PostalCode...  
1    http://nyc.pediacities.com/Resource/PostalCode...  
2    http://nyc.pediacities.com/Resource/PostalCode...  
3    http://nyc.pediacities.com/Resource/PostalCode...  
4    http://nyc.pediacities.com/Resource/PostalCode...  
..                                                 ...  
257  http://nyc.pediacities.com/Resource/PostalCode...  
258  http://nyc.pediacities.com/Resource/PostalCode...  
259  http://nyc.pediacities.com/Resource/PostalCode...  
260  http://nyc.pediacities.com/Resource/PostalCode...  
261  http://nyc.pediacities.com/Resource/PostalCode...  

[262 rows x 11 columns]

In [44]:
uhf_zipcode_crosswalk_df = pd.read_csv('uhf_zipcode_crosswalk.csv')
uhf_zipcode_crosswalk_df

Fips        Borough                              NTA_Name  \
0    101          Bronx               Kingsbridge - Riverdale   
1    102          Bronx                       Northeast Bronx   
2    103          Bronx                  Fordham - Bronx Park   
3    104          Bronx                  Pelham - Throgs Neck   
4    105          Bronx                     Crotona - Tremont   
5    106          Bronx              High Bridge - Morrisania   
6    107          Bronx              Hunts Point - Mott Haven   
7    201       Brooklyn                            Greenpoint   
8    202       Brooklyn       Downtown - Heights - Park Slope   
9    203       Brooklyn    Bedford Stuyvesant - Crown Heights   
10   204       Brooklyn                         East New York   
11   205       Brooklyn                           Sunset Park   
12   206       Brooklyn                          Borough Park   
13   207       Brooklyn              East Flatbush - Flatbush   
14   208       Brooklyn                  Canarsie - Flatlands   
15   209       Brooklyn               Bensonhurst - Bay Ridge   
16   210       Brooklyn         Coney Island - Sheepshead Bay   
17   211       Brooklyn               Williamsburg - Bushwick   
18   301      Manhattan           Washington Heights - Inwood   
19   302      Manhattan  Central Harlem - Morningside Heights   
20   303      Manhattan                           East Harlem   
21   304      Manhattan                       Upper West Side   
22   305      Manhattan                       Upper East Side   
23   306      Manhattan                     Chelsea - Clinton   
24   307      Manhattan           Gramercy Park - Murray Hill   
25   308      Manhattan              Greenwich Village - SoHo   
26   309      Manhattan        Union Square - Lower East Side   
27   310      Manhattan                       Lower Manhattan   
28   401         Queens            Long Island City - Astoria   
29   402         Queens                           West Queens   
30   403         Queens                  Flushing - Clearview   
31   404         Queens                 Bayside - Little Neck   
32   405         Queens              Ridgewood - Forest Hills   
33   406         Queens                         Fresh Meadows   
34   407         Queens                      Southwest Queens   
35   408         Queens                               Jamaica   
36   409         Queens                      Southeast Queens   
37   410         Queens                              Rockaway   
38   501  Staten Island                         Port Richmond   
39   502  Staten Island                Stapleton - St. George   
40   503  Staten Island                           Willowbrook   
41   504  Staten Island             South Beach - Tottenville   

                                              Zipcode  
0                                        10463, 10471  
1                          10466, 10469, 10470, 10475  
2                                 10458, 10467, 10468  
3            10461, 10462, 10464, 10465, 10472, 10473  
4                                 10453, 10457, 10460  
5                                 10451, 10452, 10456  
6                          10454, 10455, 10459, 10474  
7                                        11211, 11222  
8                   11201, 11205, 11215, 11217, 11231  
9                   11213, 11212, 11216, 11233, 11238  
10                                       11207, 11208  
11                                       11220, 11232  
12                         11204, 11218, 11219, 11230  
13                         11203, 11210, 11225, 11226  
14                                11234, 11236, 11239  
15                                11209, 11214, 11228  
16                         11223, 11224, 11229, 11235  
17                                11206, 11221, 11237  
18                  10031, 10032, 10033, 10034, 10040  
19                  10026, 10027, 10030, 10037, 10039  
20                              